lib/eqaf.mli

```diff
@@ -1,2 +1,5 @@
 val equal : string -> string -> bool
 (** Constant time equal function on [string]. *)
+
+val compare_be : string -> string -> int
+val compare_le : string -> string -> int
```

Comment:
would be nice to go backwards from the expected semantics (which i do not yet understand), namely:
- `equal` is well defined over the entire domain -- `equal "foo" ""` will take the same time as `equal "" ""` (AFAICT), and report `false`
- `compare_X` - not sure: it raises early an invalid_argument exception if the provided strings are of different length, is throwing an invalid argument something to be expected from compare?

now, if we have compare, I can think of using it in some map/set data structure -- here maybe not all keys are of the same length, is this an issue? or should instead compare on a short string and a long string always return 1 (or -1) -- not violating the constant time property?